File: CS/IntAsEnumExample.Module/BusinessObjects/SampleObject.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Xpo;
using DevExpress.Persistent.Base;

namespace IntAsEnumExample.Module.BusinessObjects {
    [DefaultClassOptions]
    public class SampleObject : BaseObject {
        public SampleObject(Session session) : base(session) { }
        private string name;
        public string Name {
            get { return name; }
            set { SetPropertyValue("Name", ref name, value); }
        }
        private int integerProperty;
        public int IntegerProperty {
            get { return integerProperty; }
            set { SetPropertyValue("IntegerProperty", ref integerProperty, value); }
        }
    }

    public enum SampleEnum { Value1, Value2, Value3}
}

File: VB/IntAsEnumExample.Module/BusinessObjects/SampleObject.vb
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Xpo
Imports DevExpress.Persistent.Base

Namespace IntAsEnumExample.Module.BusinessObjects

    <DefaultClassOptions>
    Public Class SampleObject
        Inherits BaseObject

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub

        Private nameField As String

        Public Property Name As String
            Get
                Return nameField
            End Get

            Set(ByVal value As String)
                SetPropertyValue("Name", nameField, value)
            End Set
        End Property

        Private integerPropertyField As Integer

        Public Property IntegerProperty As Integer
            Get
                Return integerPropertyField
            End Get

            Set(ByVal value As Integer)
                SetPropertyValue("IntegerProperty", integerPropertyField, value)
            End Set
        End Property
    End Class

    Public Enum SampleEnum
        Value1
        Value2
        Value3
    End Enum
End Namespace
